### ISSUE TITLE
addStyle function: Refused to apply inline style because it violates the following Content Security Policy

### ISSUE THREAD

**Issue by jsonjuri** [tinymce/tinymce#6378]
When using TinyMCE in combination with CSP, the addStyle function (mceDefaultStyles) will be rejected. 
At the moment there is no way to turn off the addStyle function. Currently have added custom CSS with settings.content_style that should replace addStyle inline CSS to meet the CSP policy. Still it's trying to add the CSS via addStyle.

**TinyMCE version:**
4.9.11

**Browser:**
Chrome Version 89.0.4356.6 (Official Build) dev (64-bit)

**CSP Settings:**
_`Content-Security-Policy: style-src-elem 'self' 'nonce-M2ViMTJkNGM1YTRjMjUxMDJjMDFiMjVjNTk2MzY5YWNlYmM1MWM5Ng=='_

`tinymce.js:6733 Refused to apply inline style because it violates the following Content Security Policy directive: "style-src-elem 'self' https://cdn.jsdelivr.net 'nonce-M2ViMTJkNGM1YTRjMjUxMDJjMDFiMjVjNTk2MzY5YWNlYmM1MWM5Ng=='". Either the 'unsafe-inline' keyword, a hash ('sha256-yTCJFvBsJ3q3wf4Dk4paMpnG1N2ABmaPEXkImUtBFdM='), or a nonce ('nonce-...') is required to enable inline execution.`

**There can be 2 solutions, to add a setting to turn off the contentStyles:**

![image](https://user-images.githubusercontent.com/11409905/103473409-0ad37300-4dd3-11eb-986f-1b6005f5054f.png)

**Or the ability to add a nonce to the addStyle function.**

![image](https://user-images.githubusercontent.com/11409905/103473481-28eda300-4dd4-11eb-828d-73ab36b21ccd.png)

I hope one of the above fixes can be added, or else other workarounds would be would be much appreciated.

**Comment by lnewson** [comment]
Hi, unfortunately TinyMCE 4.x is no longer supported (see https://www.tiny.cloud/blog/tinymce-4-support-window/). TinyMCE 5 is our supported community version, I'd suggest upgrading and seeing if that fixes your issue. I'm going to close this since it's not a version we are supporting, but if you can reproduce on a supported version then please let us know and we'll reopen this issue.

**Comment by jsonjuri** [comment]
@lnewson ok that's unfortunate, any chance it can be pulled and published as 4.9.12 when I solve and push the fix for this issue?
Anyway, below is a sample how to reproduce the issue.

```
<!DOCTYPE html>
<html lang="en">
    <head>
        <meta http-equiv="Content-Security-Policy" content="
            default-src 'self';
            script-src 'self' https://cdn.jsdelivr.net https://cdn.tiny.cloud 'nonce-M2ViMTJkNGM1YTRjMjUxMDJjMDFiMjVjNTk2MzY5YWNlYmM1MWM5Ng==';
            script-src-elem 'self' https://cdn.jsdelivr.net https://cdn.tiny.cloud 'nonce-M2ViMTJkNGM1YTRjMjUxMDJjMDFiMjVjNTk2MzY5YWNlYmM1MWM5Ng==';
            script-src-attr 'self' https://cdn.jsdelivr.net;
            style-src 'self' https://cdn.jsdelivr.net https://cdn.tiny.cloud;
            style-src-elem 'self' https://cdn.jsdelivr.net https://cdn.tiny.cloud 'nonce-M2ViMTJkNGM1YTRjMjUxMDJjMDFiMjVjNTk2MzY5YWNlYmM1MWM5Ng==';
            style-src-attr 'self' 'unsafe-inline';
            img-src * blob: data:;
            font-src 'self' https://cdn.jsdelivr.net https://cdn.tiny.cloud data:;
            connect-src 'self' blob: https://cdn.jsdelivr.net https://cdn.tiny.cloud;
        ">

        <!--<script src="https://cdn.tiny.cloud/1/no-api-key/tinymce/5/tinymce.min.js" referrerpolicy="origin"></script>-->
        <script src="https://cdn.jsdelivr.net/npm/tinymce@%5E4.9.11/tinymce.js"></script>
    </head>

    <body>
        <script nonce="M2ViMTJkNGM1YTRjMjUxMDJjMDFiMjVjNTk2MzY5YWNlYmM1MWM5Ng==">
            tinymce.init({
                selector: '#mytextarea'
            });
        </script>

        <h1>TinyMCE CSP Inline Style Refused</h1>
        <form method="post">
            <textarea id="mytextarea">Hello! Check the console to see the CSP error.</textarea>
        </form>

        <div style="margin: 10px 0;padding:10px;border-radius:3px 3px 3px 3px;color:#D8000C;background-color:#FFBABA;">
            tinymce.js:6733 Refused to apply inline style because it violates the following Content Security Policy directive:
            "style-src-elem 'self' https://cdn.jsdelivr.net https://cdn.tiny.cloud https://pro.fontawesome.com
            'nonce-M2ViMTJkNGM1YTRjMjUxMDJjMDFiMjVjNTk2MzY5YWNlYmM1MWM5Ng=='". Either the 'unsafe-inline' keyword, a hash
            ('sha256-yTCJFvBsJ3q3wf4Dk4paMpnG1N2ABmaPEXkImUtBFdM='), or a nonce ('nonce-...') is required to enable inline
            execution.
        </div>
    </body>
</html>

```
